databaseutil/databaseutil.go

```diff
@@ -2,82 +2,78 @@ package databaseutil
 
 import (
 	"database/sql"
-	// Notice that we’re loading the driver anonymously.
-	// The driver registers itself as being available
-	// to the database/sql package.
+	"github.com/atmiguel/cerealnotes/models/user"
 	_ "github.com/lib/pq"
-	"golang.org/x/crypto/bcrypt"
 	"log"
+	"os"
 	"time"
 )
 
 var db *sql.DB
 
-func Connect(dbUrl string) error {
-	var err error
+func init() {
```

Comment:
I'm not the biggest fan of the init doing all this db connection. I also except this will make testing more difficult in case you ever want to pass in a mock database. But I agree the way we had it before with connect was also bad.

Reply:
fair enough. any suggestions for what to do instead? perhaps while trying to write tests for this i'll discover something that makes sense here?

Reply:
I suspect creating our own db wrapper object might make sense. So you call connect and it returns a "DBConnectionHelper" or something like that. Then we change all the functions to methods of that type/class. Then in the code have main.go initialize a instance config which gets passed around. 

 Another crazy idea is having the config object initialized in main.go that stores function pointers, instead of like static info. So you can do like config.GetConnection() and it will return a db connection. That way its super easy to create mock types, because you'll just pass a new config object, with constructors for whatever type you want. .

Reply:
Remove init and recreate Connect function

Reply:
`authenticateUserCredentials`